### DISCUSSION TITLE
Run on boot for non-root devices, also question about the liscence.

### DISCUSSION THREAD

**Discussion by metalex201** [RikkaApps/Shizuku#494] [Ideas]
I've figured out a way to modify adb settings so that wireless adb could be turned on automatically by an app.

theres a permission that is required 

`<uses-permission android:name="android.permission.WRITE_SECURE_SETTINGS"
        tools:ignore="ProtectedPermissions" />`
        
its a "system" permission but it can be granted through ADB and with that (with my beginner knowledge of shizuku) it could be granted through shizuku aswell.

after that we could make a boot receiver that runs `Settings.Global.putInt(getContentResolver(), "adb_wifi_enabled", 1);` and runs shizuku afterwards

If shizuku has been previously linked to adb it should be able to reconnect after a reboot if adb is turned on. So this would make adding an experimental "Run on Boot for non-Root" option possible (also adb auth timeout can also be disabled)

Would this be possible to add? The only thing I'm stuck with is enabling the permission from the manager itself, If i can get some help with that I could make a PR to add this feature.

-----------------------------------------------------------------------

Also a different thing i would like to question is how the Liscence works, Apache 2.0 allows distribution but on the readme, it is declared that a given person is FORBIDDEN from distributing a modified version of the apk. I would like to make a version of shizuku dedicated to vr headset (specifically on quest 2/3, scoped storage has been a big issue with these devices and shizuku is a way to fix it temporarily) But the OS of these headsets handles the settings UI differently leading to shizuku opening up the vr settings app that has no developer options, which could lead to confusion for someone that wants to setup Shizuku, and I have figured out a fix for this but it would conflict with normal phone users, so a modified version for vr that has all credits to the original creator would be the best option. Am I allowed to make a fork of this project that is used in this way? Or would that be against the text declared in the readme?

**Comment by RikkaW** [comment]
You can do whatever you want, unless you declare the ownership.

About `"android.permission.WRITE_SECURE_SETTINGS"` permission, I believe that today's Google Play will never allow apps using such permissions. I even think update on Google Play will trigger a remove with a random reason. 😵

So, if you want to make a PR, please add the ability to build a non-play version, then anything you want to do is OK.
